File: CS/XPO/NonPersReloadEF/NonPersReloadEF.Module/BusinessObjects/PersistentObjects.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.BaseImpl.EF;

namespace NonPersistentObjectsDemo.Module.BusinessObjects {

    [DefaultClassOptions]
    [DevExpress.ExpressApp.DC.XafDefaultProperty(nameof(Order.Address))]
    public class Order : BaseObject {
        public virtual DateTime Date { get; set; }
        public virtual string Address { get; set; }
        public virtual decimal Total { get; set; }
        public virtual ObservableCollection<OrderLine> Details { get; set; } = new ObservableCollection<OrderLine>();
        public virtual OrderStatus Status { get; set; }
    }

    public enum OrderStatus {
        Active, Delayed
    }

    [DevExpress.ExpressApp.DC.XafDefaultProperty(nameof(OrderLine.Product))]
    public class OrderLine : BaseObject {
        public virtual Order Order { get; set; }

        public virtual Product Product { get; set; }

        public virtual int Quantity { get; set; }
    }

    [DefaultClassOptions]
    public class Product : BaseObject {
        public virtual string Name { get; set; }
        public virtual decimal Price { get; set; }
    }

}

File: VB/EFCore/NonPersReloadEF/NonPersReloadEF.Module/BusinessObjects/PersistentObjects.vb
Imports System.Collections.ObjectModel
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl.EF

Namespace NonPersistentObjectsDemo.Module.BusinessObjects

    <DefaultClassOptions>
    <DevExpress.ExpressApp.DC.XafDefaultProperty(NameOf(Order.Address))>
    Public Class Order
        Inherits BaseObject

        Public Overridable Property [Date] As Date

        Public Overridable Property Address As String

        Public Overridable Property Total As Decimal

        Public Overridable Property Details As ObservableCollection(Of OrderLine) = New ObservableCollection(Of OrderLine)()

        Public Overridable Property Status As OrderStatus
    End Class

    Public Enum OrderStatus
        Active
        Delayed
    End Enum

    <DevExpress.ExpressApp.DC.XafDefaultProperty(NameOf(OrderLine.Product))>
    Public Class OrderLine
        Inherits BaseObject

        Public Overridable Property Order As Order

        Public Overridable Property Product As Product

        Public Overridable Property Quantity As Integer
    End Class

    <DefaultClassOptions>
    Public Class Product
        Inherits BaseObject

        Public Overridable Property Name As String

        Public Overridable Property Price As Decimal
    End Class
End Namespace
